drivers/hopwa_caper/app/models/hopwa_caper/generators/fy2024/enrollment_filters/prior_living_situation_filter.rb

```diff
@@ -0,0 +1,107 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module HopwaCaper::Generators::Fy2024::EnrollmentFilters
+  PriorLivingSituationFilter = Struct.new(:label, :types, :rental_subsidy_types, keyword_init: true) do
+    def apply(scope)
+      return scope.where(prior_living_situation: [nil, 8, 9, 99]) if types == :unknown
+
+      scope = scope.where(rental_subsidy_type: rental_subsidy_type_codes) if rental_subsidy_types
+      scope.where(prior_living_situation: codes)
+    end
+
+    def rental_subsidy_type_codes
+      lookup = HudUtility2024.rental_subsidy_types.invert
+      rental_subsidy_types.map do |type|
+        lookup.fetch(type)
+      end
+    end
+
+    def codes
+      lookup = HudUtility2024.situations_for(:prior).invert
+      types.map do |type|
+        lookup.fetch(type)
+      end
+    end
+
+    def self.all
+      filters = [
+        new(
+          label: 'A place not meant for human habitation?',
+          types: ['Place not meant for habitation (e.g., a vehicle, an abandoned building, bus/train/subway station/airport or anywhere outside)'],
+        ),
+        new(
+          label: 'An emergency shelter?',
+          types: ['Emergency shelter, including hotel or motel paid for with emergency shelter voucher, or Host Home shelter'],
+        ),
+        new(
+          label: 'A transitional housing facility for formerly homeless persons?',
+          types: ['Transitional housing for homeless persons (including homeless youth)'],
+        ),
+        new(
+          label: 'A permanent housing situation for formerly homeless persons?',
+          types: [
+            'Rental by client, with ongoing housing subsidy',
+          ],
+          rental_subsidy_types: [
+            'VASH housing subsidy',
+            'Permanent Supportive Housing',
+            'Other permanent housing dedicated for formerly homeless persons',
+          ],
```

Comment:
seems like rental subsidy could be used this way, could use another set of eyes

Reply:
I'm not clear on what the goal is, can you explain a bit more?
It does seem like you can limit enrollments to those with a particular prior living situation and rental subsidy type, if that's all you mean.

Reply:
Oh sure, so the question is "A permanent housing situation for formerly homeless persons" which doesn't correspond a prior-living-situation code. But maybe we can infer it with a combination of code 435 (Rental by client, with ongoing housing subsidy) and certain values of RentalSubsidyType (3.917.A). 

Perhaps we just can't answer it though and it should be left as null